src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -185,24 +185,25 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		/* extract index key information from the index's pg_index info */
 		indexInfo = BuildIndexInfo(indexDesc);
 
-		/* only check unique indexes */
-		if (indexInfo->ii_Unique == false)
+		/* only check unique indexes and exclusion constraints. */
+		if (indexInfo->ii_Unique == false && indexInfo->ii_ExclusionOps == NULL)
```

Comment:
Hey @robin900 thanks for your contribution!

>  However, it will be more laborious to check that the exclusion operator for the distribution column is indeed '=': there are lots of equality operators in pg_operator, varying by argument data type, which would be valid for Citus to accept, and some of which have an oprname != '='.

In #748 you mentioned the above, which I agree.  

But the good news is that in Citus, we already have a function called [OperatorImplementsEquality(opno](https://github.com/citusdata/citus/blob/6333f9ba6f2c6af605ca1528764283e44e7efb15/src/backend/distributed/planner/multi_logical_planner.c#L2046)) which is exactly what you need. Given an operator number, returns true if it is an equality operator.

 I think it'd be convinient if we do something like the following in the following loop:
```C

		for (attributeIndex = 0; attributeIndex < attributeCount; attributeIndex++)
		{
			AttrNumber attributeNumber = attributeNumberArray[attributeIndex];
			if (indexInfo->ii_Unique && distributionColumn->varattno == attributeNumber)
			{
				hasDistributionColumn = true;
				break;
			}
    
			if (indexInfo->ii_ExclusionOps != NULL && 
			    distributionColumn->varattno == attributeNumber && 
			    OperatorImplementsEquality((indexInfo->ii_ExclusionOps [attributeIndex]))
			{
				hasDistributionColumn = true;
				break;
			}
		}
```
(Well, the above is just a psudo-code, but hope that it shows what I wanted to explain. We could probably write the code with nested `if` statements where the top level is`if distributionColumn->varattno == attributeNumber`)